Feat: Linux Installation Script

reisaraujo-miguel opened the pull request:
Here is only the feature code. There are some changes Godot did to the export_presets.cfg and project.godot, but I can revert this before you merge if you find it better.

reisaraujo-miguel committed:
feat: add desktop integration

This commit adds desktop integration for linux by moving the executable
to .local/godots.app and creating a desktop file in
.local/share/applications. This way we can have a system shortcut to
easily launch the application

reisaraujo-miguel committed:
chore: fix several erros and warnings

The changes focus on adding proper type annotations and fixing various static typing issues throughout the codebase. The commit handles type safety improvements and GDScript best practices.

I have also commented some unused or unreachable code, fixed narrowing conversion warnings by explicitly converting them, and added some annotations to ignore warnings from code I can't find a way  to fix (for example, the await calls from some async functions. If I leave the await keyword, I get a warning saying it is redundant, but if I remove it, I get an error saying I have to add it... wtf)

Also, we have some formatting because I have enabled gdformat to run automatically when I save a file. I think I will make a commit just with formatting to the rest of the files.

reisaraujo-miguel committed:
chore: don't create desktop entry when launching from editor

reisaraujo-miguel committed:
Fix: handle debug builds and fix some vriable naming and comments

reisaraujo-miguel committed:
Revert "chore: fix several erros and warnings"

This reverts commit be959ccb5dd62e9d3d84d348130700caffc2eed1.

MakovWait commented:
Yes please, only related code. Thanks!

reisaraujo-miguel committed:
Reverting non feature related changes

reisaraujo-miguel commented:
Done.

MakovWait commented:
Thanks! Now lets talk about the feature.

I am not sure it is OK to have some sort of side effect during startup. I think this behavior should be triggered by the user, for instance via special button in Settings tab (near Setting Folder button or as a separate settings category)

also, the button/category should be available for Linux users only

what do you think?

reisaraujo-miguel committed:
Revert "Fix: handle debug builds and fix some vriable naming and comments"

This reverts commit 41f28a6afffbbc808e20282422c87f25aa570401.

reisaraujo-miguel committed:
Revert "chore: don't create desktop entry when launching from editor"

This reverts commit 05698c0e5105cb87fadb0af1df59f347f8580d36.

reisaraujo-miguel committed:
Revert "chore: fix several erros and warnings"

This reverts commit be959ccb5dd62e9d3d84d348130700caffc2eed1.

reisaraujo-miguel committed:
Revert "feat: add desktop integration"

This reverts commit a720bdb87ca8d6740ed442d0c1252ad565117b80.

reisaraujo-miguel committed:
Fix: Reverting everything and addig an installation script

I think instead of adding a desktop file creation feature inside of the app, we could just add an installation script the user can execute. The same way bun, deno and kitty do.

reisaraujo-miguel committed:
Revert "Reverting non feature related changes"

This reverts commit ff6c9528436a94406fef685c37668245156b02f3.

reisaraujo-miguel committed:
Revert "Reverting all the desktop integration code"

This reverts commit 61160dd3acf28bfe70dd4fa64fb68a8c77b3fa96.

reisaraujo-miguel committed:
Revert "feat: add desktop integration"

This reverts commit a720bdb87ca8d6740ed442d0c1252ad565117b80.

reisaraujo-miguel committed:
Revert "chore: fix several erros and warnings"

This reverts commit be959ccb5dd62e9d3d84d348130700caffc2eed1.

reisaraujo-miguel committed:
Revert "Fix: handle debug builds and fix some vriable naming and comments"

This reverts commit 41f28a6afffbbc808e20282422c87f25aa570401.

reisaraujo-miguel committed:
Revert "chore: don't create desktop entry when launching from editor"

This reverts commit 05698c0e5105cb87fadb0af1df59f347f8580d36.

reisaraujo-miguel committed:
Merge branch 'main' into feat/desktop-integration

reisaraujo-miguel commented:
> Thanks! Now lets talk about the feature.
> 
> I am not sure it is OK to have some sort of side effect during startup. I think this behavior should be triggered by the user, for instance via special button in Settings tab (near Setting Folder button or as a separate settings category)
> 
> also, the button/category should be available for Linux users only
> 
> what do you think?

I think you're right. I tried to look at the source code to understand how to add a toggle to the configuration popup to create the desktop file. So I couldn't really figure out how the UI is built, because of that I decide to go through another route. I added an installation script to the project and added instructions to the README on how to use it. That's the same way bun, deno and kitty do installation.

reisaraujo-miguel committed:
add exit conditions and fix desktop exec and icon paths

novadragonDOTspace commented:
Code needs to check if unzip is installed in the first place. Arch Linux as a major one doesnt default install it in the first place.

reisaraujo-miguel committed:
Handle edge cases and update desktop database after install

This commit handles a few edge cases for downloading files and
unzipping. I used the python3 method as the default for unzipping
because I think most systems should have python installed, since it is a
dependacy for a lot widely used applications.

I also added some lines to update the desktop database after installing.
This was added mostly because of my personal Hyprland configuration,
where it doesn't do this automatically, although on most DE's like Gnome
or Plasma doing this would be unnecessary.

reisaraujo-miguel committed:
fix typo

MakovWait commented on install.sh:
```diff
@@ -0,0 +1,91 @@
+
+printf "Downloading Godots...\n"
+try_download "$GODOTS_DOWNLOAD_URL" "$GODOTS_TMP_DIR"
+
```
shasum check should be added
sha256 may be fetched from `digest property at https://api.github.com/repos/{owner}/{repo}/releases` [source](https://github.com/ScoopInstaller/Scoop/issues/6381)

also I added `SHA512-SUMS.txt` to the latest workflow [example](https://github.com/MakovWait/godots/releases/tag/v1.4.dev2)

zibetnu commented on install.sh:
```diff
@@ -0,0 +1,91 @@
+GODOTS_DOWNLOAD_URL="https://github.com/MakovWait/godots/releases/latest/download/LinuxX11.zip"
+GODOTS_TMP_DIR="/tmp/godots"
+
+GODOTS_INSTALL_DIR="$HOME/.local/godots.app"
```
To throw in my 2 cents, what's up with this install directory? I think it would make more sense to default to  `$HOME/.local/share/godots` but prefer  `$XDG_DATA_HOME/godots` if that variable exists.

reisaraujo-miguel committed:
Merge remote-tracking branch 'upstream/main' into feat/desktop-integration

reisaraujo-miguel committed:
chore: add install script instruction on README

reisaraujo-miguel committed:
fix: fix gtk-update-icon-cache command

reisaraujo-miguel committed:
feat: verifies checksum and adds some error handleling

reisaraujo-miguel commented:
Hey! I just made some changes to the script following your suggestions, and also added some error handling to the commands in case they fail during the execution.
